main/src/modules/reactiveLayer/sensationManager/src/opcSensation.cpp

```diff
@@ -133,16 +133,80 @@ Bottle OpcSensation::handleEntities()
             else if (entity->entity_type() == "object" && dynamic_cast<Object*>(entity)->m_present == 1.0) {  // Known entities and present!
                 known_obj = true;
                 addToEntityList(k_entities, entity->entity_type(), entity->name());
+                Object* obj1 = dynamic_cast<Object*>(entity);
+                //Handle red balls
+                if (entity->name() == "red_ball"){
+                    yDebug("Found a redBall object");
+                    // Set color:
+                    obj1->m_color[0] = 250;
+                    obj1->m_color[1] = 0;
+                    obj1->m_color[2] = 0;
+                    if (!Network::isConnected("/pf3dTracker/data:o",("/"+moduleName+"/pf3dTracker:i").c_str())){
+                        yDebug("RedBall Port not connected...");
+                        obj1->m_present = 0.0;
+                    }else{
+                        yDebug("RedBall Port connected...\nUpdateing position...");
+                        //update position
+                        Bottle *bot = pf3dTrackerPort.read();
+                        yDebug("RedBall Port readed");
+                        obj1->m_ego_position[0] = bot->get(0).asDouble();
+                        obj1->m_ego_position[1] = bot->get(1).asDouble();
+                        obj1->m_ego_position[2] = bot->get(2).asDouble();
+                        obj1->m_dimensions[0] = bot->get(3).asDouble();
+                        obj1->m_dimensions[1] = bot->get(4).asDouble();
+                        obj1->m_dimensions[2] = bot->get(5).asDouble();
+                        obj1->m_present = bot->get(6).asDouble();
+                        iCub->opc->commit(obj1);
+                        yDebug("opc updated...");
+                    }
+                }
+                //Object* obj1 = iCub->opc->addOrRetrieveEntity<Object>(entity->name());
+                if (obj1->m_present == 1.0){
+                    if (entity->name() == "red_ball"){ 
```

Comment:
Because there is nothing in the `if` besides the comment "//change color to red". So if it needs to be changed, please change the color, otherwise get rid of the `if` condition :)